feat(ref): add pickup points to delivery (#73)
ulule/developers.ulule.com: content/reference/delivery.md

## Delivery resource

| Field                   | Type                                                    | Description                                                                                        |
| ----------------------- | ------------------------------------------------------- | -------------------------------------------------------------------------------------------------- |
| `address_required`      | bool                                                    | Whether backers must specify a shipping address when ordering a reward                             |
| `date_delivery`         | string                                                  | Estimated date of delivery of rewards, with the YYYY-MM-DD format                                  |
| `has_shippings`         | bool                                                    | Whether there are shipping costs                                                                   |
| `phone_number_required` | bool                                                    | Whether backers must specify the phone number in the shipping address when ordering a reward       |

> | `pickup_points`         | array of [pickup point resouce](#pickup-point-resource) | Pick up points, only present in the project delivery resource                                      |

Reviewer: ```suggestion
| `pickup_points`         | array of [pickup point resources](#pickup-point-resource) | Pickup points, only present in the project delivery resource                                      |
```

> | `pickup_points`         | array of [pickup point resouces](#pickup-point-resource) | Pick up points, only present in the project delivery resource                                      |

Reviewer: ```suggestion
| `pickup_points`         | array of [pickup point resources](#pickup-point-resource) | Pickup points, only present in the project delivery resource                                      |
```
It would be great if you could double check review comments, it's the second time I send this comment

| `id`        | int              | Unique ID of the shipping                                                                                                                            |
| `zone`      | string           | [Shipping zone](#shipping-zones)                                                                                                                     |

> ### Pickup point resource

Reviewer: move below shipping zones subsection

| Field         | Type                 | Description                                      |
| ------------- | -------------------- | ------------------------------------------------ |

> | `city`        | string               | City                                             |

Reviewer: sort fields alphabetically

| `address`     | string               | Address                                          |

> | `description` | [i18n object](#i18n) | Description                                      |

Author: or maybe `details` ?

Reviewer 2: `description` is ok

| `id`          | int                  | Unique ID of the pickup point                    |

> | `order_count` | int                  | Count of order with the pickup point as delivery |

Reviewer: do we need it? if not, remove

Author: Need to ask @payetgil

| `update_shippings` | array of [update-shipping payloads](#update-shipping-payload) | Update shippings -- the shipping `id` field must be set                          |
| `delete_shippings` | array of ints                                                 | Delete shippings -- the ints in the array must be existing shippings `id` fields |

> It's possible to create, update, and delete project pickup points with the following payload fields.

Reviewer: ```suggestion
Similarily, project pickup points may be created, updated, and deleted with the following payload fields:
```

| Field                  | Type                                                                  | Description                                                                             |
| ---------------------- | --------------------------------------------------------------------- | --------------------------------------------------------------------------------------- |
| `create_pickup_points` | array of [create-pickup-point payloads](#create-pickup-point-payload) | Create pickup points                                                                    |

> | `update_pickup_points` | array of [update-pickup_point payloads](#update-pickup-point-payload) | Update pickup points -- the shipping `id` field must be set                             |

Reviewer: description is wrong

| `amount`    | float            | Shipping cost -- can be between 1 and 999.99        |
| `countries` | array of strings | Countries, represented by their two-letter ISO code |
| `zone`      | string           | [Shipping zone](#shipping-zone)                     |

### Create pickup point payload

> | Field         | Type                 | Description |

Reviewer: can you double check the spec for required fields and length limits? and also double check review comments, it's the second time I send this comment.

| ------------- | -------------------- | ----------- |

> | `address`     | string               | Address     |

Reviewer: if we have an i18n field in the resource, shouldn't we have an i18n field in the payload? same for the city field and the update pickup point payload

| `amount`    | float            | Shipping cost -- can be between 1 and 999.99        |
| `countries` | array of strings | Countries, represented by their two-letter ISO code |
| `zone`      | string           | [Shipping zone](#shipping-zone)                     |

### Create pickup point payload

> | Field         | Type                 | Description                        |

Reviewer: which fields are required? how are fields validated?

| ------------- | -------------------- | ---------------------------------- |

> | `city`        | string               | City                               |

Reviewer: sort fields alphabetically

| `address`     | string               | Address                            |
| `description` | [i18n object](#i18n) | Description                        |

### Update pickup point payload

| Field         | Type                 | Description                        |
| ------------- | -------------------- | ---------------------------------- |

> | `city`        | string               | City                               |

Reviewer: sort fields alphabetically
